rero_ils/modules/contributions/api.py

```diff
@@ -244,28 +252,58 @@ def get_authorized_access_point(self, language):
             language=language
         )
 
-    def update_online(self, dbcommit=False, reindex=False):
+    def documents_pids(self, with_subjects=True):
+        """Get documents pids."""
+        doc_pids = []
+        query = DocumentsSearch() \
+            .filter('term', contribution__agent__pid=self.pid)
+        for hit in query.source('pid').scan():
+            doc_pids.append(hit.pid)
+        if with_subjects:
+            query = DocumentsSearch() \
+                .filter('term', subjects__pid=self.pid) \
+                .filter(
+                    'terms', subjects__type=['bf:Person', 'bf:Organisation'])
+            for hit in query.source('pid').scan():
+                doc_pids.append(hit.pid)
+        return list(set(doc_pids))
```

Comment:
Maybe you could do only one search by combine filters with something like :
```python
search_filters = Q("term", contribution__agent__pid=self.pid)

if with_subjects:
    subject_filters = Q("term", subjects__pid=self.pid) & \
                      Q("terms", subjects__type=['bf:Person', 
                                                 'bf:Organisation'])
    search_filters = search_filters | subject_filters

search = DocumentsSearch() \
    .query('bool', filter=[search_filters]) \
    .source('pid')

return [hit.pid for hit in search.scan()]
```
see: 
https://elasticsearch-dsl.readthedocs.io/en/latest/search_dsl.html#:~:text=Query%20objects%20can,python%22)%0A%23%20%7B%22bool%22%3A%20%7B%22must_not%22%3A%20%5B...%5D%7D%7D